### ISSUE TITLE
Console view doesn't show all builds for a commit

### ISSUE THREAD

**Issue by aj062** [buildbot/buildbot#3945]
On a large buildbot instance having lot of builds in parallel, console view is almost unusable.

Console view shows only 2-3 recent commits, and even for those it shows very few builds, it doesn't show all the builds corresponding to those commits. In my instance, I have 200+ builds running at the same time, and many of them are not related to commits (e.g.: Nightly schedulers).

This bug happens always, and make console view unusable, as it show very few builds for a commit (and most of the builds are missing).

This happens because console view fetches just 200 recent builds (by default). If there are 200+ builds running at that time, this wouldn't even fetch all the on-going builds, let alone completed builds. Increasing the default value of 200 in settings help little bit, but it still doesn't guarantee that all the builds corresponding to a commit would be shown.

A better implementation of console view would be to fetch last few changes/commits and fetch all the builds corresponding to those commits. It would need an API "to fetch all builds for a revision." as discussed in https://github.com/buildbot/buildbot/issues/3927

This would make console view more reliable and show all the builds for recent commits, and would not be affected by any other kinds of builds (e.g.: nightly) running on Buildbot.

**Comment by uglycoyote** [comment]
I agree -- similarly, I have tried to implement a "more..." button on the bottom of the console page, but found it very difficult due to fact that it's so difficult to associate builds with changes.  My implementation would request the next several hundred builds each time you press it, and then do a separate loop to make sure we had requested all of the changes corresponding to the builds that had been received. 

 It would be much cleaner if the change structure contained id's of the builds which were associated with it, then console could do firstly a request for the desired number of changes, the secondly a request for the relevant set of builds (perhaps by posting a query with the desired buildid's)

**Comment by mojca** [comment]
@p12tic, @rajdeepbharati: This is also relevant for the GSOC project as we are facing the same problem.

**Comment by p12tic** [comment]
@mojca: Should I add it to the GSOC project?
